runtime/sam/op/distinct/distinct.go

```diff
@@ -0,0 +1,54 @@
+package distinct
+
+import (
+	"encoding/binary"
+
+	"github.com/brimdata/super"
+	"github.com/brimdata/super/runtime/sam/expr"
+	"github.com/brimdata/super/zbuf"
+)
+
+type Op struct {
+	parent zbuf.Puller
+	expr   expr.Evaluator
+	cache  []byte
+	block  map[string]struct{}
+}
+
+func New(parent zbuf.Puller, expr expr.Evaluator) *Op {
+	return &Op{
+		parent: parent,
+		expr:   expr,
+		cache:  make([]byte, 0, 1024),
+	}
+}
+
+func (o *Op) Pull(done bool) (zbuf.Batch, error) {
+	if o.block == nil {
+		o.block = make(map[string]struct{})
+	}
+	for {
+		batch, err := o.parent.Pull(done)
+		if batch == nil || err != nil {
+			o.block = nil
```

Comment:
Nit: Could use `clear` and initialize `o.block` in `New`.
```suggestion
			clear(o.block)
```